core/dbt/task/freshness.py

```diff
@@ -35,10 +36,41 @@ def on_skip(self):
 
     def before_execute(self):
         description = 'freshness of {0.source_name}.{0.name}'.format(self.node)
-        print_start_line(description, self.node_index, self.num_nodes)
+        fire_event(PrintStartLine(description=description,
+                                  index=self.node_index,
+                                  total=self.num_nodes))
 
     def after_execute(self, result):
-        print_freshness_result_line(result, self.node_index, self.num_nodes)
+        if hasattr(result, 'node'):
+            source_name = result.node.source_name
+            table_name = result.node.name
+        else:
+            source_name = result.source_name
+            table_name = result.table_name
+        if result.status == FreshnessStatus.RuntimeErr:
+            fire_event(PrintHookEndErrorLine(source_name=source_name,
+                                             table_name=table_name,
+                                             index=self.node_index,
+                                             total=self.num_nodes,
+                                             execution_time=result.execution_time))
+        elif result.status == FreshnessStatus.Error:
+            fire_event(PrintHookEndErrorStaleLine(source_name=source_name,
+                                                  table_name=table_name,
+                                                  index=self.node_index,
+                                                  total=self.num_nodes,
+                                                  execution_time=result.execution_time))
+        elif result.status == FreshnessStatus.Warn:
+            fire_event(PrintHookEndWarnLine(source_name=source_name,
+                                            table_name=table_name,
+                                            index=self.node_index,
+                                            total=self.num_nodes,
+                                            execution_time=result.execution_time))
+        else:
+            fire_event(PrintHookEndPassLine(source_name=source_name,
+                                            table_name=table_name,
+                                            index=self.node_index,
+                                            total=self.num_nodes,
+                                            execution_time=result.execution_time))
```

Comment:
this looks like it was pretty non-trivial to change. I really like what you did with it here: It brings the important stuff to the top, and buries the stuff we don't care about as much. Nice job.